weather-app/src/app/app.component.ts

```diff
@@ -27,6 +42,20 @@ export class AppComponent implements OnInit {
     });
   }
 
+  toggleTheme(): void {
+    if (this.isDarkMode === false) {
```

Comment:
Think of making it more generalized here.

Instead of two pretty similar `if(){...}` blocks you can try and do it in one shot:
``
this.isDarkMode = !this.isDarkMode;
this.this.localStorageThemeService.setTheme(this.isDarkMode === true ? 'dark' :'light');
``